File: daily/2026/3090_Maximum_Length_Substring_With_Two_Occurrences.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    int maximumLengthSubstring(string s) {
        int n = s.length();
        int start = 0;
        int ans = 0;
        vector<int> cnt(26);
        for (int end = 0; end < n; end++){
            int curr = s[end] - 'a';
            cnt[curr]++;
            while(start <= end && cnt[curr] > 2){
                cnt[s[start++] - 'a']--;
            }
            ans = max(ans, end - start + 1);
        }
        return ans;
    }
};

int main(){
    string s = "bcbbbcba";
    // string s = "aaaa";
    Solution S;
    cout << S.maximumLengthSubstring(s) << endl;
    return 0;
}

File: daily_python/3090_Maximum_Length_Substring_With_Two_Occurrences.py
import os
import sys
from typing import List

class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        n = len(s)
        ans = 0
        start = 0
        cnt = [0] * 26
        for end in range(n):
            curr = ord(s[end]) - ord('a')
            cnt[curr] += 1
            while cnt[curr] > 2:
                cnt[ord(s[start]) - ord('a')] -= 1
                start += 1
            ans = max(ans, end - start + 1)
        return ans

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            s = lines[0].split('=')[1].strip()[1:-2]
            dataset.append(s)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for s in dataset:
        results.append(solution.maximumLengthSubstring(s))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')